tests/integration/ha_tests/helpers.py

```diff
@@ -231,16 +240,25 @@ async def are_writes_increasing(
     ops_test, down_unit: str = None, extra_model: Model = None
 ) -> None:
     """Verify new writes are continuing by counting the number of writes."""
-    writes, _ = await count_writes(ops_test, down_unit=down_unit, extra_model=extra_model)
-    for member, count in writes.items():
-        for attempt in Retrying(stop=stop_after_delay(60 * 3), wait=wait_fixed(3), reraise=True):
-            with attempt:
-                more_writes, _ = await count_writes(
-                    ops_test, down_unit=down_unit, extra_model=extra_model
-                )
-                assert (
-                    more_writes[member] > count
-                ), f"{member}: writes not continuing to DB (current writes: {more_writes[member]} - previous writes: {count})"
+    for attempt in Retrying(stop=stop_after_delay(60), wait=wait_fixed(3), reraise=True):
+        with attempt:
+            writes, _ = await count_writes(ops_test, down_unit=down_unit, extra_model=extra_model)
+            assert len(writes), "No units report count"
+    logger.info(f"Initial writes {writes}")
+    for attempt in Retrying(stop=stop_after_delay(60 * 3), wait=wait_fixed(3), reraise=True):
+        with attempt:
+            more_writes, _ = await count_writes(
+                ops_test, down_unit=down_unit, extra_model=extra_model
+            )
+            logger.info(f"Retry writes {more_writes}")
+            members_checked = []
+            for member, count in writes.items():
+                if member in more_writes:
+                    members_checked.append(member)
+                    assert (
+                        more_writes[member] > count
+                    ), f"{member}: writes not continuing to DB (current writes: {more_writes[member]} - previous writes: {count})"
+            assert len(members_checked), "No member checked from the initial writes"
```

Comment:
Count writes is ignoring the async replica, so we can't assume the the members will match if calling during a topology change.